### PR TITLE
Register existing store with new apiBase when re-registering Kube API

### PR DESCRIPTION
**Pull request by nevalla**
This PR fix issue when api version is resolved from Kubernetes API and respective store has been already registered with default api version.

Signed-off-by: Lauri Nevala <lauri.nevala@gmail.com>

### CODE REVIEW

File: src/renderer/api/kube-api.ts [hunk 1]
```diff
@@ -229,8 +229,11 @@ export class KubeApi<T extends KubeObject = any> {
       });
 
       if (this.apiVersionPreferred) {
-        Object.defineProperty(this, "apiBase", { value: this.getUrl() });
+        const apiBase = this.getUrl();
+
+        Object.defineProperty(this, "apiBase", { value: apiBase });
         apiManager.registerApi(this.apiBase, this);
+        apiManager.updateStoreKey(this.objectConstructor.apiBase);
```
Comment (ixrock) [inline src/renderer/api/kube-api.ts]: Maybe better to do all the job for updating the keys inside `registerApi` since we pass `this` already?
Something like: 

```ts
apiManager.registerApi(this.apiBase, this, {update: true} /*options={}*/);
```

Also, purpose of `updateStoreKey` is not clear for me (why / when should I use it).

Comment (jakolehm) [inline src/renderer/api/kube-api.ts]: Maybe we could do the store key update automatically when `registerApi` is called?

File: src/renderer/api/kube-api.ts [hunk 2]
```diff
@@ -229,7 +229,9 @@ export class KubeApi<T extends KubeObject = any> {
       });
 
       if (this.apiVersionPreferred) {
-        Object.defineProperty(this, "apiBase", { value: this.getUrl() });
+        const apiBase = this.getUrl();
```
Comment (jakolehm) [inline src/renderer/api/kube-api.ts]: Unnecessary change, please revert 🙂

**Review by jakolehm — DISMISSED** [review]

File: src/renderer/api/kube-api.ts (hunk 1 above)
Comment (ixrock) [reply src/renderer/api/kube-api.ts]: This is what I also proposed.

File: src/renderer/api/api-manager.ts [hunk 3]
```diff
@@ -23,6 +23,12 @@ export class ApiManager {
 
   registerApi(apiBase: string, api: KubeApi) {
     if (!this.apis.has(apiBase)) {
+      this.stores.forEach((store) => {
+        if(store.api === api) {
+          this.stores.set(apiBase, store);
+        }
+      });
+
```
Comment (ixrock) [inline src/renderer/api/api-manager.ts:31]:
```suggestion
      const store = this.getStore(api);
      if (store) this.stores.set(apiBase, store);
```

**Review by ixrock — APPROVED** [review]

File: src/renderer/api/api-manager.ts (hunk 3 above)
Comment (nevalla) [reply src/renderer/api/api-manager.ts:31]: This does not work since api has updated value for `apiBase` and no stores are found with that key.

File: src/renderer/api/api-manager.ts [hunk 4]
```diff
@@ -23,6 +23,12 @@ export class ApiManager {
 
   registerApi(apiBase: string, api: KubeApi) {
     if (!this.apis.has(apiBase)) {
+      this.stores.forEach((store) => {
```
Comment (jakolehm) [inline src/renderer/api/api-manager.ts:26]: Test for this would be nice.

**Review by jakolehm — APPROVED** [review]

File: src/renderer/api/api-manager.ts (hunk 3 above)
Comment (ixrock) [reply src/renderer/api/api-manager.ts:31]: Ok, but sounds like a bug since `apiManager.getStore(api: string | KubeApi)`..
Introduced from here: https://github.com/lensapp/lens/pull/2091
